lib/u3d/cache.rb

```diff
@@ -51,13 +51,18 @@ def [](key)
       @cache[key]
     end
 
-    def initialize(path: nil, force_os: nil, force_refresh: false)
+    def initialize(path: nil, force_os: nil, force_refresh: false, offline: false)
+      raise "Cache: cannot specify both offline and force_refresh" if offline && force_refresh
       @path = path || default_path
       @cache = {}
       os = force_os || U3dCore::Helper.operating_system
       Utils.ensure_dir(@path)
       file_path = File.expand_path(DEFAULT_NAME, @path)
       need_update, data = check_for_update(file_path, os)
+      if offline
+        UI.verbose("Cache outdated but we are working offline, so no updating it.")
+        need_update = false
+      end
```

Comment:
This will raise errors when the cache is not yet present as `data` will be `nil`.

```ruby
if !File.file?(file_path)
  need_update = true
else
```

Reply:
offline has overriding. I can't update if I am offline

I think I had modified 

```ruby
def [](key)
    return nil if @cache[key].nil?
    @cache[key]
end
```

to be

```ruby
def [](key)
    return nil @cache.nil`
    return nil if @cache[key].nil?
    @cache[key]
end
```
for that case but removed it by accident. I would need to add some tests.

Reply:
OK. How would we interpret `latest` then? Fail and log a message?

Reply:
Maybe if `need_update` is true and `offline` is true then we should hard fail. `user_error!`

Reply:
I need to add tests anyway.

Reply:
Tests added!